bmp085.js

```diff
@@ -147,36 +147,42 @@ BMP085.prototype.readWord = function (register, length, callback) {
 };
 
 BMP085.prototype.calibrate = function () {
-    this.calibrationData = {};
-    this.waitForCalibrationData();
-
     var self = this;
-
-    this.calibrationRegisters.forEach(function(register) {
-        self.readWord(register, function(reg, value) {
-            self.calibrationData[reg.name] = value;
+    this.calibrationData = {};
+    self.events.on('calibrating', function () {
```

Comment:
Should this maybe be moved to the `waitForCalibrationData` function instead of using an event? When using the event, are we guaranteed that this function completes before the `for` loop (line 168) runs?

Reply:
yea more refactoring might be welcome... I will try this out.